internal/bminventory/inventory.go

```diff
@@ -5560,7 +5560,24 @@ func (b *bareMetalInventory) GetClusterByKubeKey(key types.NamespacedName) (*com
 }
 
 func (b *bareMetalInventory) GetHostByKubeKey(key types.NamespacedName) (*common.Host, error) {
-	return b.hostApi.GetHostByKubeKey(key)
+	h, err := b.hostApi.GetHostByKubeKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	//customize the host as done in the REST API
```

Comment:
Maybe add all the customize part inside the `if`? And handle all errors
```
	//customize the host as done in the REST API
	if h.ClusterID != nil {
		cluster, err = common.GetClusterFromDB(b.db, *h.ClusterID, common.SkipEagerLoading)
		if err != nil {
			return nil, fmt.Errorf("can not find a cluster for host %s", h.ID.String())
		}
  	       err = b.customizeHost(&cluster.Cluster, &h.Host)
		if err != nil {
			return nil, fmt.Errorf("failed to customize host %s", h.ID.String())
		}
	}
	return h, nil
```

Reply:
No need. Event if the cluster is nil customize is still working on the host